network/peers/conn_manager.go

```diff
@@ -74,6 +76,7 @@ func (c connManager) TrimPeers(ctx context.Context, logger *zap.Logger, net libp
 		if !c.connManager.IsProtected(pid, protectedTag) {
 			err := net.ClosePeer(pid)
 			logger.Debug("closing peer", zap.String("pid", pid.String()), zap.Error(err))
+			metrics.DeletePeerInfo(pid)
```

Comment:
this means we only evict when we're the ones closing the connection, but not the other way around, so the metric might keep growing indefinitely

i guess the right place to put this is in `conn_handler.go` in the disconnection callback, unless we'd like to remember peers for a bit after we've disconnected (for debugging), in which case a more complex eviction mechanism is needed